plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableSearchQuery.php

```diff
@@ -164,7 +164,7 @@ private function generate_where(): string {
 
 		$where_statement = implode( ' OR ', $where );
 
-		return " ( $where_statement ) ";
+		return $where_statement ? " ( $where_statement ) " : '';
```

Comment:
The result of the call to `$this->generate_where_for_search_filter()` is filterable and so, while the current solution covers the scenario you've described, I think it may be worth doing something more explicit than testing if the string is truthy (because something like `(string) ' '` will still pass the test, yet could still lead to the same problem of an empty where clause).

What do you think of changing the earlier block (beginning line 159) from:

```php
$search_where = $this->generate_where_for_search_filter( $search_filter );
if ( ! empty( $search_where ) ) {
	$where[] = $search_where;
}
```

To:

```php
$search_where = trim( $this->generate_where_for_search_filter( $search_filter ) );
if ( strlen( $search_where ) > 0 ) {
	$where[] = $search_where;
}
```

Similarly, on this line:

```suggestion
		return ( strlen( $where_statement ) > 0 ) ? " ( $where_statement ) " : '';
```

We could be similarly explicit in `OrdersTableQuery::build_query()`.